src/server/aws.ts

```diff
@@ -111,11 +116,54 @@ export const storeS3File = async (
 }
 
 export async function signedUrlForFile(Key: string) {
-    return await getSignedUrl(getS3Client(), new GetObjectCommand({ Bucket: s3BucketName(), Key }), {
+    return await getSignedUrl(getS3BrowserClient(), new GetObjectCommand({ Bucket: s3BucketName(), Key }), {
         expiresIn: 3600,
     })
 }
 
+export const signedUrlForCodeUpload = async (jobInfo: MinimalJobInfo) => {
+    const prefix = pathForStudyJobCode(jobInfo)
+
+    return await createPresignedPost(getS3BrowserClient(), {
+        Bucket: s3BucketName(),
+        Conditions: [['starts-with', '$key', prefix]],
+        Expires: 3600,
+        Key: prefix + '/${filename}', // single quotes are intentional, S3 will replace ${filename} with the filename
+    })
+}
+
+export const signedUrlForStudyFileUpload = async (path: string) => {
+    return await createPresignedPost(getS3BrowserClient(), {
+        Bucket: s3BucketName(),
+        Expires: 3600,
+        Conditions: [['starts-with', '$key', path]],
+        Key: path + '/${filename}', // single quotes are intentional, S3 will replace ${filename} with the filename
+    })
+}
+
+export const deleteFolderContents = async (folderPath: string) => {
+    const listCommand = new ListObjectsV2Command({
+        Bucket: s3BucketName(),
+        Prefix: folderPath,
+    })
+
+    const listedObjects = await getS3Client().send(listCommand)
+
+    if (!listedObjects.Contents) {
+        console.error('No objects found in the folder.')
+        return
+    }
+
+    const objectsToDelete = listedObjects.Contents.map(({ Key }) => ({ Key }))
+
+    const deleteCommand = new DeleteObjectsCommand({
+        Bucket: s3BucketName(),
+        Delete: { Objects: objectsToDelete },
+    })
+
+    await getS3Client().send(deleteCommand)
```

Comment:
I think we should add a limit to the amount of files that can be deleted, just to ensure that the code doesn't mistakenly pass `/studies` into this and delete EVERYTHING.  something like this maybe?
```suggestion
    if (objectsToDelete.length > 10) { throw new Error('cowardly refusing to delete more than 10 files at once') }
    await getS3Client().send(deleteCommand)
```